kafka-connect-hazelcast/src/main/scala/com/datamountaineer/streamreactor/connect/hazelcast/writers/HazelCastWriter.scala

```diff
@@ -78,8 +71,8 @@ class HazelCastWriter(client: HazelcastInstance, settings: HazelCastSinkSettings
       logger.debug("No records received.")
     } else {
       logger.debug(s"Received ${records.size} records.")
-      val converted = records.map(r => (r.topic, convert(r)))
-      converted.foreach({
+      val grouped = records.map(r => (r.topic, r))
+      grouped.foreach({
         case (topic, payload) =>
           val writer = writers.get(topic)
           val t = Try(writer.foreach(w => w.write(payload)))
```

Comment:
Try(writer.foreach(w => w.write(payload)))  => if w.write throws an exception we will swallow it. because of that we don't instruct the framework to retry and it will commit the offsets and we lose data